lib/workers/repository/process/vulnerabilities.ts

```diff
@@ -0,0 +1,120 @@
+import { Ecosystem, Osv, OsvOffline } from '@jamiemagee/osv-offline';
+import pAll from 'p-all';
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type { PackageRule, RenovateConfig } from '../../../config/types';
+import { logger } from '../../../logger';
+import type {
+  PackageDependency,
+  PackageFile,
+} from '../../../modules/manager/types';
+
+export class Vulnerabilities {
+  private osvOffline: OsvOffline;
+
+  private static readonly managerEcosystemMap: Record<string, Ecosystem> = {
```

Comment:
```suggestion
  private static readonly managerEcosystemMap: Record<string, Ecosystem | undefined> = {
```

Reply:
I'm not sure it can be undefined? It's a constant map.

Reply:
It's undefined for unsupported ecosystems 🤔

Reply:
Is it? You would wanna set a property for each manager, even if there us no equivalent Ecosystem?

Reply:
Yes, we pass the manager name to that dictionary and validate the return value below. So because we pass any manager we get undefined for any not supported value.

Reply:
Wouldn't it be more appropriate to make this a Partial Record?